test/addons-abi/test_instanceof/test1.js

```diff
@@ -0,0 +1,344 @@
+// Copyright 2010 the V8 project authors. All rights reserved.
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are
+// met:
+//
+//     * Redistributions of source code must retain the above copyright
+//       notice, this list of conditions and the following disclaimer.
+//     * Redistributions in binary form must reproduce the above
+//       copyright notice, this list of conditions and the following
+//       disclaimer in the documentation and/or other materials provided
+//       with the distribution.
+//     * Neither the name of Google Inc. nor the names of its
+//       contributors may be used to endorse or promote products derived
+//       from this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+// "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+// LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+// A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+// OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+// SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+// LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+// DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+// THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+// (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+// Adapted from the V8 test suite for the instanceof operator to test
+// whether it is possible to implement instanceof using NAPI.
+
+require('../../common');
+var assert = require('assert');
+var addon = require('./build/Release/test_instanceof');
+
+function assertTrue(assertion) {
+	return assert.strictEqual(true, assertion);
+}
+
+function assertEquals(leftHandSide, rightHandSide) {
+	assert.equal(leftHandSide, rightHandSide);
+}
+
+var except = "exception";
+
+var correct_answer_index = 0;
+var correct_answers = [
```

Comment:
Since I copied this test from V8, I decided to change as little as possible. In fact, I'm contemplating loading it dynamically and replacing `x instanceof y ` with `addon.doInstanceOf(x, y)`. I'd prefer this to be part of a well-documented and independently maintained ECMAScript test suite, but alas.